CorsixTH/Lua/queue.lua

```diff
@@ -321,11 +321,14 @@ function Queue:movePatient(index, new_index)
   self:move(first_patient_index + index - 1, new_index)
 end
 
---! Called when reception desk is destroyed, or when a room is destroyed from a crashed machine.
-function Queue:rerouteAllPatients(action)
+--! Called when reception desk is destroyed (or unstaffed), or when a room is destroyed from a crashed machine.
+--!param action (function) What should the patient do now
+--!param origin The room or object, if needed
+function Queue:rerouteAllPatients(action, origin)
   for _, humanoid in ipairs(self) do
     -- check by class type as staff/vips shouldn't get a SeekRoomAction
-    if class.is(humanoid, Patient) then
+    -- Except at reception desks, a Vip could otherwise get stuck
+    if class.is(humanoid, Patient) or (class.is(humanoid, Vip) and origin.class == "ReceptionDesk") then
```

Comment:
VIPs don't actually get queued when visiting rooms (might need to check what happens with a build a room on path scenario). They get a walk to the door, but don't actually queue so they only ever get the expected callback.
Thus technically we don't need to check they are at a reception desk. We also could just use the action.name to determine if we were after a reception desk or a room or doing something else.

Reply:
If that's the case the code will work fine without the origin.class == "ReceptionDesk" ?